src/test/ui/issues/issue-41726.nll.stderr

```diff
@@ -3,6 +3,8 @@ error[E0596]: cannot borrow data in a `&` reference as mutable
    |
 LL |         things[src.as_str()].sort(); //~ ERROR cannot borrow immutable
    |         ^^^^^^^^^^^^^^^^^^^^ cannot borrow as mutable
+   |
+   = help: trait `IndexMut` is required to modify indexed content, but it is not implemented for `std::collections::HashMap<std::string::String, std::vec::Vec<std::string::String>>`
```

Comment:
I think in those cases it might be best to check if `IndexMut` is implemented and if it is then keep just this part of the message:

```
   |
   = help: trait `IndexMut` is required to modify indexed content
``` 

Doing this, the error message wouldn't be wrong and that note is still helpful information which could be the correct approach for the user to take; if they specifically want `Index` and not `IndexMut` then they'll know that and can ignore the suggestion.

Reply:
I've pushed a new commit that implements what I've suggested above and added a new test that shows this. If we don't want to take that approach then we can change it.